xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1285,24 +1444,36 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
 
   // Adds a scalar DAmax return value to an FP8 GEMM.
   absl::Status F8AddDAmax(HloInstruction *instr, HloInstruction *existing_gemm,
-                          HloInstruction *reduce_damax) {
+                          HloInstruction *reduce_damax,
+                          HloInstruction *get_tuple_element) {
     // Change the output shape of the Custom Call to tuple(D, DAmax).
     Shape damax_shape = ShapeUtil::MakeScalarShape(F32);
-    Shape tuple_shape =
-        ShapeUtil::MakeTupleShape({instr->shape(), damax_shape});
+
+    std::vector<Shape> tuple_shape_vec = {instr->shape(), damax_shape};
+    if (get_tuple_element) {
+      tuple_shape_vec.insert(tuple_shape_vec.begin() + 1,
+                             existing_gemm->shape().tuple_shapes(1));
```

Comment:
I don't think this works if the tuple output has more than 2 outputs. You should handle an arbitrary number of outputs by copying the tuple shape, replacing the first shape with `instr->shape()`, and appending damax_shape.